faraday/data.py

```diff
@@ -0,0 +1,246 @@
+path = ''
+
+
+for location in os.curdir, relpath1, relpath2, setuppath, userpath:
+    try:
+        logging.config.fileConfig(os.path.join(location, "loggingConfig.ini"))
+        path = location
+        break
+    except ConfigParser.NoSectionError:
+        pass
+
+logger = logging.getLogger('Data')
+
+# Load Telemetry Configuration from telemetry.ini file
+
+#Create Proxy configuration file path
+dataConfigPath = os.path.join(path, "data.ini")
+logger.debug('data.ini PATH: ' + dataConfigPath)
+
+dataConfig = ConfigParser.RawConfigParser()
+dataConfig.read(dataConfigPath)
+
+# Command line input
+parser = argparse.ArgumentParser(description='Provides a generic data server to proxy.')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Data configuration file')
+parser.add_argument('--flaskhost', help='Set Flask server hostname/address')
+parser.add_argument('--flaskport', help='Set Flask server port')
+parser.add_argument('--proxyhost', help='Set Proxy server hostname/address')
+parser.add_argument('--proxyport', help='Set Proxy server port')
+parser.add_argument('--start', action='store_true', help='Start Data server')
+
+# Parse the arguments
+args = parser.parse_args()
+
+
+def initializeDataConfig():
+    '''
+    Initialize Data configuration file from data.sample.ini
+
+    :return: None, exits program
+    '''
+
+    logger.info("Initializing Data")
+    shutil.copy(os.path.join(path, "data.sample.ini"), os.path.join(path, "data.ini"))
+    logger.info("Initialization complete")
+    sys.exit(0)
+
+
+def configureData(args, simpleuiConfigPath):
+    '''
+    Configure Data configuration file from command line
+
+    :param args: argparse arguments
+    :param SimpleUIConfigPath: Path to simpleui.ini file
+    :return: None
+    '''
+
+    config = ConfigParser.RawConfigParser()
+    config.read(os.path.join(path, "data.ini"))
+
+    if args.flaskhost is not None:
+        config.set('FLASK', 'HOST', args.flaskhost)
+    if args.flaskport is not None:
+        config.set('FLASK', 'PORT', args.flaskport)
+    if args.proxyhost is not None:
+        config.set('PROXY', 'HOST', args.proxyhost)
+    if args.proxyport is not None:
+        config.set('PROXY', 'PORT', args.proxyport)
+
+    with open(simpleuiConfigPath, 'wb') as configfile:
+        config.write(configfile)
+
+
+# Now act upon the command line arguments
+# Initialize and configure Data
+if args.init:
+    initializeDataConfig()
+
+configureData(args, dataConfigPath)
+
+# Read in configuration file settings
+dataConfig.read(dataConfigPath)
+
+# Check for --start option and exit if not present
+if not args.start:
+    logger.warning("--start option not present, exiting Data server!")
+    sys.exit(0)
+
+host = dataConfig.get("FLASK", "HOST")
+port = dataConfig.get("FLASK", "PORT")
+
+# Global variables
+packet_struct = struct.Struct('2B 40s')
+PACKET_LEN = 42
+PAYLOAD_LEN = 40
+
+
+# Definitions
+APP_RFDATAPORT_UART_PORT = 1
```

Comment:
This should be in the configuration file. Probably need to update telemetry for this too eh...